### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by Stephen-Adom**
# Blog App: validations and model specs
In this pull request, the following tasks were implemented:
  - Added the following validations:
     - For the User model:
         - Name must not be blank.
         - PostsCounter must be an integer greater than or equal to zero.
     - For the Post model:
         - Title must not be blank.
         - Title must not exceed 250 characters.
         - CommentsCounter must be an integer greater than or equal to zero.
         - LikesCounter must be an integer greater than or equal to zero.
 - Added unit specs for all model methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject { Post.new(author_id: 1, title: 'title', text: 'text', comments_counter: 0, likes_counter: 0) }
+
+  before { subject.save }
+
+  it 'new post should be saved in the database' do
```
Comment (t-yanick) [inline spec/models/post_spec.rb:8]: - [ ] As per project requirements, your are to add units tests for all your models. At the moment you have tests just for the user and post model. I understand that you don't have any validations in the comment and like models, but you can at least add the test of new comment and new line being saved in the database.
image
image

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  it 'comments_counter should be an integer' do
+    subject.comments_counter = 'a'
+    expect(subject).to_not be_valid
+  end
+
+  it 'comments_counter should be greater than or equal to 0' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (t-yanick) [inline spec/models/post_spec.rb:35]: - [ ] Great job working on this milestone. Your post_spec is currently missing the following validation test cases:
- Title must not exceed 250 characters.
- LikesCounter must be an integer 
- LikesCounter should be greater than or equal to zero.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Stephen-Adom , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All required model validations correctly added
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Use of correct Gitflow
:heavy_check_mark: No linter errors

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Stephen-Adom , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All required model validations correctly added
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Use of correct Gitflow
:heavy_check_mark: No linter errors

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 1 above)
Comment (peterrobert) [reply spec/models/post_spec.rb:8]: This has been done well

File: spec/models/post_spec.rb (hunk 2 above)
Comment (peterrobert) [reply spec/models/post_spec.rb:35]: This tests has been done well

**Review by peterrobert — APPROVED** [review]
Hi @Stephen-Adom, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work making the changes requested by the previous reviewer. ✔️

### OPTIONAL

- It is always good practice to capitalize the first letter of your commit messages. This will make your project even more professional


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**